src/Resources/ResourceManager/SdkModels/Resources/PSResource.cs

```diff
@@ -77,96 +77,92 @@ public string TagsTable
 
         public string ETag { get; set; }
 
-        public PSResource(GenericResource resource)
+        public PSResource(string id, string subscriptionId, string resourceGroupName, string name, string kind, string location, string type, Identity identity, PSObject properties, Plan plan, Sku sku, IDictionary<string, string> tags)
         {
-            this.ResourceId = resource.Id;
-            this.Id = resource.Id;
-            this.Identity = resource.Identity;
-            this.Kind = resource.Kind;
-            this.Location = resource.Location;
-            this.ManagedBy = resource.ManagedBy;
-            this.ResourceName = resource.Name;
-            this.Name = resource.Name;
-            this.Plan = resource.Plan;
-            this.Properties = ((JToken)resource.Properties).ToPsObject();
-            this.ResourceType = resource.Type;
-            this.Sku = resource.Sku;
-            this.Tags = resource.Tags;
-            this.Type = resource.Type;
-
-            var resourceIdentifier = new ResourceIdentifier(this.Id);
-            this.ParentResource = resourceIdentifier.ParentResource;
-            this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+            this.ResourceId = id;
+            this.Id = id;
+            this.SubscriptionId = subscriptionId;
+            this.ResourceGroupName = resourceGroupName;
+            this.ResourceName = name;
+            this.Name = name;
+            this.Kind = kind;
+            this.Location = location;
+            this.ResourceType = type;
+            this.Type = type;
+            this.Identity = identity;
+            this.Properties = properties;
+            this.Plan = plan;
+            this.Sku = sku;
+            this.Tags = tags;
         }
 
-        public PSResource(GenericResourceExpanded resource)
+        public PSResource(GenericResource resource): this(
+            resource.Id,
+            (new ResourceIdentifier(resource.Id)).Subscription, 
+            (new ResourceIdentifier(resource.Id)).ResourceGroupName,
```

Comment:
If we add `ParentResource` here would be 3 `new`s. I'm thinking we should move in down in the constructor body.